examples/http/server.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+#include "opentelemetry/ext/http/server/http_server.h"
+#include<string>
+#include<atomic>
+
+
+namespace {
+
+class HttpServer : public HTTP_SERVER_NS::HttpRequestCallback
+{
+
+protected:
+    HTTP_SERVER_NS::HttpServer server_;
+    std::string server_url_ ;
+    uint16_t port_ ;
+    std::atomic<bool> is_running_{false};
+
+public:
+
+    HttpServer(std::string server_name = "test_server",uint16_t port = 8800): port_(port){
+        server_.setServerName(server_name);
+        server_.setKeepalive(false);
+    }
+
+    void AddHandler(std::string path, HTTP_SERVER_NS::HttpRequestCallback *request_handler){
+        server_.addHandler(path, *request_handler);
+    }
+
+    void Start() {
+        if (!is_running_) {
+            server_.addListeningPort(port_);
+            server_.start();
+            is_running_ = true;
+        }
+    }
+
+    void Stop() {
+        if (is_running_){
```

Comment:
Same issue. Would be best to use [exchange](https://en.cppreference.com/w/cpp/atomic/atomic/exchange)

Reply:
fixed this.